### PR TITLE
Fix isWritable throwing RangeError on Array "length"

### PR DESCRIPTION
**Pull request by shvaikalesh**

### CODE REVIEW

File: harness/propertyHelper.js [hunk 1]
```diff
@@ -140,7 +140,8 @@ function isSameValue(a, b) {
 }
 
 function isWritable(obj, name, verifyProp, value) {
-  var newValue = value || "unlikelyValue";
+  var unlikelyValue = Array.isArray(obj) && name === "length" ? 217493 : "unlikelyValue";
```
Comment (ljharb) [inline harness/propertyHelper.js]: rather than modifying this helper, why not pass in the value from the test?

Comment (shvaikalesh) [reply harness/propertyHelper.js]: The test uses convenient `verifyProperty` helper, which doesn't allow passing a `value`.

Comment (ljharb) [reply harness/propertyHelper.js]: Perhaps that would be a better helper change than this one?

Comment (shvaikalesh) [reply harness/propertyHelper.js]: As a test author, I have an expectation that both `verifyWritable` and `verifyProperty` would be as less intrusive/observable as possible and would not throw any non-related errors.

It is a bit hard to achieve with Array length, as setting `length` may delete elements, so this change now uses `2 ** 32 - 1` for _unlikely_ length.

File: harness/propertyHelper.js [hunk 2]
```diff
@@ -191,7 +194,7 @@ function verifyNotWritable(obj, name, verifyProp, value) {
     assert(!Object.getOwnPropertyDescriptor(obj, name).writable,
          "Expected obj[" + String(name) + "] to have writable:false.");
   }
-  if (isWritable(obj, name, verifyProp)) {
```
Comment (shvaikalesh) [inline harness/propertyHelper.js]: `value` was lost here.

File: harness/propertyHelper.js [hunk 3]
```diff
@@ -139,8 +139,11 @@ function isSameValue(a, b) {
   return a === b;
 }
 
-function isWritable(obj, name, verifyProp, value) {
-  var newValue = value || "unlikelyValue";
+function isWritable(obj, name, value) {
+  var unlikelyValue = Array.isArray(obj) && name === "length" ?
```
Comment (devsnek) [inline harness/propertyHelper.js]: this breaks with `test/language/global-code/decl-lex-configurable-global.js`, because it shadows `Array`.

**Pull request merged**